nats/nuid.py

```diff
@@ -35,25 +35,26 @@ def __init__(self):
         self._srand = SystemRandom()
         self._prand = Random(self._srand.randint(0, MaxInt))
         self._seq = self._prand.randint(0, MAX_SEQ)
-        self._inc = MIN_INC + self._prand.randint(0, INC)
+        self._inc = MIN_INC + self._prand.randint(BASE + 1, INC)
```

Comment:
The root problem is not in `_inc=62`, the problem is that you use non-integer division in the `next` function below. 

The program to demonstrate the problem:

    BASE = 62

    def calc(l):
        for i in range(10):
            a = int(l) % BASE
            l /= BASE
            print('{}: {} - {}'.format(i, a, l))

    calc(628556970159648696)
    print()
    calc(628556970159648758)  # 628556970159648696 + 62

Output:

    0: 18 - 1.013801564773627e+16
    1: 8 - 163516381415101.12
    2: 9 - 2637360990566.147
    3: 0 - 42538080493.00237
    4: 29 - 686098072.4677802
    5: 58 - 11066097.943028713
    6: 27 - 178485.4506940115
    7: 49 - 2878.797591838895
    8: 26 - 46.43221922320799
    9: 46 - 0.748906761664645

    0: 18 - 1.013801564773627e+16
    1: 8 - 163516381415101.12
    2: 9 - 2637360990566.147
    3: 0 - 42538080493.00237
    4: 29 - 686098072.4677802
    5: 58 - 11066097.943028713
    6: 27 - 178485.4506940115
    7: 49 - 2878.797591838895
    8: 26 - 46.43221922320799
    9: 46 - 0.748906761664645
    
If you use the integer division, as in [Go implementation](https://github.com/nats-io/nuid/blob/master/nuid.go#L108), everything works properly:
 
    0: 18 - 10138015647736269
    1: 7 - 163516381415101
    2: 9 - 2637360990566
    3: 0 - 42538080493
    4: 29 - 686098072
    5: 58 - 11066097
    6: 27 - 178485
    7: 49 - 2878
    8: 26 - 46
    9: 46 - 0

    0: 18 - 10138015647736270
    1: 8 - 163516381415101
    2: 9 - 2637360990566
    3: 0 - 42538080493
    4: 29 - 686098072
    5: 58 - 11066097
    6: 27 - 178485
    7: 49 - 2878
    8: 26 - 46
    9: 46 - 0

P.S. This should also improve performance, since integer division is more light-weight operation.
P.P.S. And this is really a critical issue, it seems to me that the fix should also be backported to the 0.11* branch.

Reply:
Moreover, I've just checked, this fix doesn't work at all. The same problem, albeit less likely to occur, appears for `_inc = 124`:

    calc(613793501461948122)
    print()
    calc(613793501461948246)  # 613793501461948122 + 124
    
    0: 0 - 9899895184870132.0
    1: 58 - 159675728788227.94
    2: 17 - 2575414980455.2896
    3: 41 - 41538951297.66596
    4: 27 - 669983085.4462252
    5: 49 - 10806178.79751976
    6: 12 - 174293.20641160905
    7: 11 - 2811.1807485743393
    8: 21 - 45.34162497700547
    9: 45 - 0.731316531887185

    0: 0 - 9899895184870132.0
    1: 58 - 159675728788227.94
    2: 17 - 2575414980455.2896
    3: 41 - 41538951297.66596
    4: 27 - 669983085.4462252
    5: 49 - 10806178.79751976
    6: 12 - 174293.20641160905
    7: 11 - 2811.1807485743393
    8: 21 - 45.34162497700547
    9: 45 - 0.731316531887185

Reply:
Thanks for the explanation and proper fix!!